enumerables.rb

```diff
@@ -0,0 +1,153 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each(&block)
+    return to_enum(:my_each) unless block_given?
+
+    each(&block)
+    self
+  end
+
+  def my_each_with_index
+    # return to_enum(:my_each_with_index) unless block_given?
+    i = 0
+    while i < length
+      yield i, self[i]
+      i += 1
+    end
+    self
+  end
+
```

Comment:
- [ ] Enumerables #my_each_with_index should return an enumerator if no block is given
- [ ] Enumerables #my_each_with_index when a block is given when self is an array behaves like #each_with_index calls the given block once for each element in self
- [ ] Enumerables #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
  
     ` NameError: undefined local variable or method `length' for 5..50:Range`
- [ ] Enumerables #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self

     ` NameError:
       undefined local variable or method `length' for 5..50:Range `

- [ ] Enumerables #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self